### ISSUE TITLE
implicit_authorization_target with "global" actions and TestController

### ISSUE THREAD

**Issue by zverok** [palkan/action_policy#277]
It is rather a request for clarification than a bug report/feature request. It might be that our environment has something wrong about it, or our understanding of the ActionPolicy usage is sub-part.

So, the situation is:
1. We have quite a few policies with some rules _not depending on the record_ (or, rather, not expecting some record to exist), like some `TeamPolicy` allows to `create?` if the current `user` is a manager in `authorization_context[:organization]` (in this case, like in many others, there is no `record` to talk about when we are deciding whether somebody is allowed to _create_ some)
2. In some view, the check is `allowed_to?(:create, with: TeamPolicy)`
3. By not being provided with a record, ActionPolicy tries to find `implicit_authorization_target`, and, in a case of view helper method, converges to `controller_name.classify.safe_constantize`. It doesn’t matter for us what it will find, we don’t use it for this check :)
4. ...But, after switching Rails 6.0→6.1, the code relying on this behavior starts to fail, because in Rails 6.0, there was `TestController` in views that had `controller_name == "test"`, while in Rails 6.1, there is a custom class for each such test, and it has `controller_name == nil` (and the code fails with `undefined method 'classify' for nil:NilClass`)

So, I can think of these possible “solutions”:
1. We are doing it wrong, there is another API for a record-less check
2. Or, there is no special API for a record-less check, but we should do it explicitly: `allowed_to?(:create, nil, with: TeamPolicy)` (this fixes the problem)
3. We are doing it wrong on the Rails testing side, there is a way to have name for a controller
4. Or, it might be considered some adjustment to ActionPolicy code: either support name-less test controlers, or, providing a convenience API for a record-less checks (if it doesn’t exist yet)

Basically, there seem to be **two** problems:
* a sane API for record-less checks (looking up _any_ record for this case is pointless)
* a sane handling of TestController (even before the controller lost its name, the `implicit_authorization_target` in this case would’ve derive `Test` as a name of the class to lookup; which would either be `nil` on `safe_constantize`, or in some codebases will fetch some probably completely unrelated class)

Or am I missing something here?..
Thanks!

**Comment by palkan** [comment]
Fixed handling nameless controllers in 0.7.2.
